main.go

```diff
@@ -146,11 +147,12 @@ func sendMetadata(ctx context.Context, apiClient *api.Client, cfg config.Config,
 	}
 }
 
-func runTestsWithRetry(testRunner TestRunner, testsCases *[]plan.TestCase, maxRetries int, mutedTest []plan.TestCase, timeline *[]api.Timeline) (runner.RunResult, error) {
+func runTestsWithRetry(testRunner TestRunner, testsCases *[]plan.TestCase, maxRetries int, mutedTests []plan.TestCase, timeline *[]api.Timeline) (runner.RunResult, error) {
 	attemptCount := 0
 
-	var testResult runner.RunResult
-	var err error
+	// load mutedTests from test plan into the runResult
+	runResult := testRunner.Result()
```

Comment:
I'm confused by the data flow.

This seems to be the only call to `Result()`, a method added to the test runner interface so that each test runner can provide ...a result? _before_ running the tests?? merely so that you can load muted tests into it???

And as a consequence of the new interface method, each test runner must not only implement it but pre-initialise a result and save it in a field, and also keep it updated?

I'd reconsider this approach - at least the `Result` method. Maybe the initial `*RunResult` should be an argument passed to `testRunner.Run`, and the runner updates that `RunResult` that it points to.

Reply:
Yes, you are right. The muted tests need to be loaded before each run so it can correctly update the result. I was trying to avoid having the main function initialize the `RunResult` struct, so I initialize it inside each test runner. But looking back at it now, I can see how it can be confusing and unnecessarily complicated. 

I have updated the implementation by passing the `*RunResult` to the `testRunner.Run` function. Let me know what you think.